src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,27 @@
-import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/store';
+import {
+  actions as filterActions,
+  filterSelector,
+} from './../../features/filter';
+
+export const TodoFilter = () => {
+  const filter = useAppSelector(filterSelector);
+  const dispatch = useDispatch();
 
-export const TodoFilter: React.FC = () => {
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            onChange={event =>
+              dispatch(filterActions.setStatus(event.target.value))
+            }
```

Comment:
use `handle` functions for all such cases

```suggestion
            onChange={handleStatusChange}
```
```
const handleStatusChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
  dispatch(filterActions.setStatus(e.target.value));
};
```